### ISSUE TITLE
Solver founds no solution though it should

### ISSUE THREAD

**Issue by castek** [albertosantini/node-quadprog#4]
Hello,
I'm using your QP. I have a set of solvable inputs. With the solver I get around 20% failure when output.Solution contains NaN. How can I tune (some threshold) the accuracy of the solver, so that I can get solution in 100% of solvable inputs?

My inputs are in format of 
minimize 1/2 * x^T D x + d^T x
where A x >= b

I'm calling:
var output = solveQP(Dmat, dvec, Amat, bvec, meq, factorized);

One example of solvable input that returns NaN:
JSON.stringify(Dmat)
"[[],[0,2,0,0,0,0,0],[0,0,2,0,0,0,0],[0,0,0,2,0,0,0],[0,0,0,0,2,0,0],[0,0,0,0,0,2,0],[0,0,0,0,0,0,2]]"
JSON.stringify(dvec)
"[0,-10,-10,-30,-20,-30,-20]"
JSON.stringify(Amat)
"[[],[0,1,-1,0,0,0,0,-1,0,0,1,0,0,0,0,0],[0,0,0,1,-1,0,0,-1,0,0,0,1,0,0,0,0],[0,0,0,1,-1,0,0,0,-1,0,0,0,1,0,0,0],[0,0,0,1,-1,0,0,0,0,-1,0,0,0,1,0,0],[0,0,0,0,0,1,-1,0,-1,0,0,0,0,0,1,0],[0,0,0,0,0,1,-1,0,0,-1,0,0,0,0,0,1]]"
JSON.stringify(bvec)
"[0,3.999,-4.001,4.999,-5.001,0.999,-1.001,-7.001,-1.001,-2.001,-0.001,-0.001,-0.001,-0.001,-0.001,-0.001]"
JSON.stringify(meq)
"0"
JSON.stringify(factorized)
"[0,0]"
This returns:
factorized = [0, 0]
output = {"solution":[,NaN,NaN,NaN,NaN,NaN,NaN],"Lagrangian":[,NaN,0,NaN,0,NaN,0,0,0,0,0,0,0,0,NaN,0],"value":[,NaN],"unconstrained_solution":[0,-4.999999999999999,-4.999999999999999,-14.999999999999996,-9.999999999999998,-14.999999999999996,-9.999999999999998],"iterations":[,6,1],"iact":[,3,5,1,14,0,0,0,0,0,0,0,0,0,0,0],"message":""}

If there is more than one possible solution with equal minimum value, it's enough to get any solution.

The constraints are of three types:
1) x2 + x3 + x4 = 5,
this returned NaNs for many inputs, so I substituted it with two inequalities:
x2 + x3 + x4 >= 4.999 and
x2 + x3 + x4 <= 5.001
This substitution helped a lot.
2) sum of some variables <= some positive integer (eg. 7 as in bvec[7])
this I substituted with:
sum of some variables <= some positive integer + epsilon
eg.
sum of some variables <= 7.001
3) each variable >= 0
that I substituted with:
each variable >= -epsilon
eg.
each variable >= -0.001

The best I would like to get only integers in solution but I'm noticed you wrote the solver that returns reals.

**Comment by albertosantini** [comment]
Did you notice the arrays are based one-index? 
See, for instance, the tests how to setup correctly the inputs.

Hope that helps you.

Otherwise I need the the js test you run to reproduce the issue and the R script to compare the results.

**Comment by castek** [comment]
Hello,
I'm not comparing it with R, I used to compare it with accord, they ported the same fortran code of goldfarb-idnani to c# as you did. I think inputs are well formed (Amat transposition, arrays starting  from 1). I tried several thousands of solvable tasks and node-quadprog finds solution only in around 80 % of cases. Accord finds solution in 97 % of my tasks. I need to find solution in 100% of solvable tasks on javascript (I used c# accord only as comparison). Here I attached the same examples in javascript as well as in R. Though amat seems to be different, they are equal in both languages, only tranformed.
[javascript_R_QP.txt](https://github.com/albertosantini/node-quadprog/files/57294/javascript_R_QP.txt)

**Comment by albertosantini** [comment]
Thanks for the details. Nice catch.

I am afraid I need to investigate in depth to find the bug.
I need to compare step by step, iteration by iteration where is the difference.

**Comment by castek** [comment]
In fact I would like to solve this "analytic" task (Dmat and dvec are the same as above):
```
Amat <- matrix(c(1,0,0,0,0,0, 0,1,1,1,0,0, 0,0,0,0,1,1, -1,-1,0,0,0,0, 0,0,-1,0,-1,0, 0,0,0,-1,0,-1, 1,0,0,0,0,0, 0,1,0,0,0,0, 0,0,1,0,0,0, 0,0,0,1,0,0, 0,0,0,0,1,0, 0,0,0,0,0,1), 6,12)
bvec <- c(4,5,1,-7,-1,-2,0,0,0,0,0,0)
solve.QP(Dmat,dvec,Amat,bvec,3)
```

```javascript
var amat = JSON.parse("[[],[0,1,0,0,-1,0,0,1,0,0,0,0,0],[0,0,1,0,-1,0,0,0,1,0,0,0,0],[0,0,1,0,0,-1,0,0,0,1,0,0,0],[0,0,1,0,0,0,-1,0,0,0,1,0,0],[0,0,0,1,0,-1,0,0,0,0,0,1,0],[0,0,0,1,0,0,-1,0,0,0,0,0,1]]");
var bvec = JSON.parse("[0,4,5,1,-7,-1,-2,0,0,0,0,0,0]");

var out = solveQP(dmat, dvec, amat, bvec,3);
```
R is not able to solve it (R says 'constraints are inconsistent, no solution!').
So I had to change the "analytical" task to "numerical":
```
Amat <- matrix(c(1,0,0,0,0,0, 0,1,1,1,0,0, 0,0,0,0,1,1, -1,-1,0,0,0,0, 0,0,-1,0,-1,0, 0,0,0,-1,0,-1, 1,0,0,0,0,0, 0,1,0,0,0,0, 0,0,1,0,0,0, 0,0,0,1,0,0, 0,0,0,0,1,0, 0,0,0,0,0,1), 6,12)
bvec <- c(4,5,1,-7.001,-1.001,-2.001,0,0,0,0,0,0)
solve.QP(Dmat,dvec,Amat,bvec,3)
```

```javascript
var amat = JSON.parse("[[],[0,1,0,0,-1,0,0,1,0,0,0,0,0],[0,0,1,0,-1,0,0,0,1,0,0,0,0],[0,0,1,0,0,-1,0,0,0,1,0,0,0],[0,0,1,0,0,0,-1,0,0,0,1,0,0],[0,0,0,1,0,-1,0,0,0,0,0,1,0],[0,0,0,1,0,0,-1,0,0,0,0,0,1]]");
var bvec = JSON.parse("[0,4,5,1,-7.001,-1.001,-2.001,0,0,0,0,0,0]");

var out = solveQP(dmat, dvec, amat, bvec,3);
```
R solves this correctly, javascript not.

So I modified it finally with other two steps:
1) equalities are replaced by pairs of inequalities with 2*epsilon tolerance
2) x(i) >= 0 are replaced by x(i) >= -epsilon
For most tasks this helped to be solved also in javascript, but around 20% of tasks these modifications are not enough. Modified task can be seen two comments above.

**Comment by albertosantini** [comment]
Please, give a try to the master, it is fixed there.
Javascript and R give the same results.

If it is ok for you, in a few hours I publish a new release.

**Comment by albertosantini** [comment]
As you saw, `fourth` and `fifth` tests reproduce the examples you provided in this issue.
Both tests have the same results as well as in R examples.

Thanks a lot for your feedback.

**Comment by castek** [comment]
100% of tasks have been solved. It works perfectly! Thanks a lot.

**Comment by castek** [comment]
`third` and `fifth` tests address this issue. `fourth` seems to be something else.

**Comment by albertosantini** [comment]
Published `1.4.0`.

Thanks a lot for the snippets.
